hydroengine_service/dgds_functions.py

```diff
@@ -0,0 +1,427 @@
+                  function=None,
+                  start_date=None,
+                  end_date=None,
+                  image_num_limit=None):
+    """
+
+    :param source: String, source/location of Earth Engine Object
+    :param dataset: String, name of Earth Engine Object dataset
+    :param image_id: String, Earth Engine Image id
+    :param band: String, band name to select from image
+    :param function: String, function to apply to image
+    :param start_date: String, start date to filter collection on
+    :param end_date: String, end date to filter collection on
+    :param image_num_limit: String, limit of objects to return
+    :return: Dictionary
+    """
+    # Get list of objects with imageId and date for collection
+    data_params = get_dgds_source_vis_params(source, image_id)
+    info = get_image_collection_info(source, start_date, end_date, image_num_limit)
+    if not info:
+        return
+
+    # get most recent to return url
+    returned_url_id = info[-1]["imageId"]
+    if image_id:
+        returned_url_id = image_id
+
+    if data_params.get('function', None) and not function:
+        function = data_params['function']
+        if isinstance(function, list):
+            function = function[0]
+        else:
+            function = function.get(band, None)
+
+    image_info = _get_wms_url(returned_url_id, type=data_params['type'], band=band, function=function)
+    image_info['dataset'] = dataset
+    image_info['band'] = band
+    image_info['imageTimeseries'] = info
+
+    return image_info
+
+def visualize_gebco(source, band):
+    """
+    Specialized function to visualize GEBCO data
+    :param source: String, Google Earth Engine image id
+    :param band: String, band of image to visualize
+    :return: Dictionary
+    """
+    data_params = DATASETS_VIS[source]
+    image = ee.Image(source)
+
+    gebco = image.select(data_params['bandNames'][band])
+    # Angle for hillshade (keep at 315 for good perception)
+    azimuth = 315
+    # Lower is longer shadows
+    zenith = 30
+
+    bathy_only = data_params.get('bathy_only', False)
+
+    height_multiplier = 30
+    # Weight between image and  hillshade (1=equal)
+    weight = 0.3
+    # make darker (<1), lighter (>1)
+    val_multiply = 0.9
+    # make  desaturated (<1) or more saturated (>1)
+    sat_multiply = 0.8
+
+    # palettes
+    # visualization params
+    topo_rgb = gebco.mask(gebco.gt(0)).visualize(**data_params['topo_vis_params'])
+    bathy_rgb = gebco.mask(gebco.lte(0)).visualize(**data_params['bathy_vis_params'])
+    image_rgb = topo_rgb.blend(bathy_rgb)
+
+    if (bathy_only):
+        # overwrite with masked version
+        image_rgb = bathy_rgb.mask(gebco.multiply(ee.Image(-1)).unitScale(-1, 10).clamp(0, 1))
+
+    # TODO:  see how this still fits in the hillshade function
+    hsv = image_rgb.unitScale(0, 255).rgbToHsv()
+
+    z = gebco.multiply(ee.Image.constant(height_multiplier))
+
+    def radians(image):
```

Comment:
This is generic and could be outside this function. It also could use a slightly longer name (deg2rad_image) and/or docstring.